values.go

```diff
@@ -38,6 +39,12 @@ func GetFlags(dir string) (map[string]string, error) {
 
 	if version, err := versionFromFile(dir); err != nil {
 		return nil, fmt.Errorf("failed to get version: %v", err)
+	} else if version == "" {
+		// ex: v0.1.2 0.1.2 0.1.2.dev3
+		matched, err := regexp.MatchString(`^v?(\d+\.){1,3}(\d+)`, gitSummary)
```

Comment:
@codeskyblue I'm getting lost here. you started by saying if there's no VERSION file, we'll use `GitSummary()` and I expected you to do so. 

Then you came up with regex and special logic. This logic is not clear to anyone from outside. It's hard to document and understand for a user. It's not a clean method to solve this problem. Therefore I cannot accept your method.

If VERSION file is missing we do _best effort_ to provide a useful string. In this case, we'll provide `GitSummary()`, which is (1) if there's a tag, uses the `tag[-commitnumber[-dirty]]` syntax (2) if not uses short commit hash.

Your regex-based approach just doesn't work:

1. when there's no tag ––the whole point of this PR to provide a string when VERSION file is missing, if there's no tag matching your regex, you fail to provide a string
2. your regex strips of important information such as v1.0-rc1 and v1.0-rc2 will just match to v1.0; which is obviously incorrect.

If you don't agree these, I'm afraid I'm not convinced in what you proposed in #4 and I have to close this PR.